First draft of UI models (#5)
jhu-sheridan-libraries/jhuda-data-models: ui/models/submission.md

# Submission

> The top level object encapsulating a logical grouping of data and metadata to be deposited to the data archive. A submission will contain one or more files, with associated file metadata, that a user intends to submit to the data archive. The user must also provide metadata describing the submission as a whole. Much of the submission data will be provided by the user throught the UI

Reviewer: This is a nit picky thing and a style point, but you might consider adding periods for the closing sentence in many of the table cells where there are two or more sentences.

| filesStatus | enum | This status summarizes the state of the files associated with this submission. [Files Status](#files-status) |
| metadata | string | A stringified JSON object with the submission/package level metadata. Must validate against the JSON schema (to be provided). Note: the metadata blob must always contain user information |
| files | List[URI] | A list of (Fedora) [File](file.md) objects |
| accessUrl | URI | A public URL where someone can view this submission. The UI should never write to this field |
| requiredActions | List[URI] | A list of references to [SubmissionActions](submissionAction.md) that must be addressed by the user |
| state |  | State for a draft submission that enables the UI to initialize views nicely. _Only_ the UI should read or write this field |

**Notes on metadata blobs**

The submission holds metadata as a stringified JSON object that must validate against our [metadata JSON schema](#). When parsed, the blob should something like:

``` javascript
{
  "user": {
    "email": "holstein@moo.org",
    "eppn": "",
    "firstName": "",
    "lastName": ""
  },
  "description": "Some text describing the submission",
  "moo": 5
}
```

> If the user reqeusts a DOI for the submission, the research will collaborate with the curator in order to get the required metadata. This interaction may happen outside of this deposit UI. If the DOI is _not_ requested, the curator will review and edit package level metadata without collaboration with the user.

Reviewer: ```suggestion
If the user requests a DOI for the submission, the researcher will collaborate with the curator in order to get the required metadata. This interaction may happen outside of this deposit UI. If the DOI is _not_ requested, the curator will review and edit package level metadata without collaboration with the user.
```

## Submission Status

A status summarizing the current state of the submission.

| Value | Description |    |
| ----- | ----------- | -- |
| draft | The submission is still user editable. This also means that submission has not yet been deposited in the archive | Submissions will always start with this status. The UI can change a submission's status _to_ `draft` only from `requiresAction` if the user has resolved all [SubmissionActions](submissionAction.md)s in `submission.requiredActions` |

> | pendingReview | Submission metadaata or files are pending review |

Reviewer: ```suggestion
| pendingReview | Submission metadata or files are pending review |
```

## Metadata Status

| Value | Description |  |
| ----- | ----------- | --- |
| draft | Submission metadata are user editable | Metadata should start in this state. If the user resolves all metadata actions, the metadata status should be changed to `draft` |
| pendingReview | Submission metadata is ready for the curator to ... curate | Should be set only after `filesStatus === approved`* by the deposit UI |
| requiresAction | One or more metadata fields have been flagged. The user must approve changes to the metadata or make changes to the metadata. This status should take precedence over other metadata statuses | This status is associated with the existence of one or more [SubmissionActions](submissionAction.md) with type `metadata` |
| approved | The metadata curator has approved the metadata. If a user makes a change to the metadata, the status must change to `draft` | This status must not exist if the submission has one or more [SubmissionActions](submissionAction.md) of type `metadata` |

> \* In our current workflow, the metadata curator will only do their curation after the honest broker approves of the set of files attached to the submission. In the workflow, the user will enter metadata, attach files, then let the honest broker that the submission is ready for review. Once the honest broker approves the attached files, the user must use the deposit UI to send the submission over to the archive, where the curator will be able to do their curation tasks.

Reviewer: ```suggestion
\* In our current workflow, the metadata curator will only do their curation after the honest broker approves of the set of files attached to the submission. In the workflow, the user will enter metadata, attach files, then let the honest broker know that the submission is ready for review. Once the honest broker approves the attached files, the user must use the deposit UI to send the submission over to the archive, where the curator will be able to do their curation tasks.
```
